src/main/java/de/tum/in/www1/artemis/domain/notification/GroupNotificationFactory.java

```diff
@@ -165,57 +194,92 @@
+                text = NEW_COURSE_POST_TEXT;
+                placeholderValues = new String[] { course.getTitle(), author.getName(), post.getTitle(), post.getContent(), post.getCreationDate().toString() };
+                notification = new GroupNotification(course, title, text, true, placeholderValues, author, groupNotificationType);
                 notification.setTransientAndStringTarget(createCoursePostTarget(post, course));
             }
             case NEW_ANNOUNCEMENT_POST -> {
                 title = NEW_ANNOUNCEMENT_POST_TITLE;
-                text = "Course \"" + course.getTitle() + "\" got a new announcement.";
-                notification = new GroupNotification(course, title, text, author, groupNotificationType);
+                text = NEW_ANNOUNCEMENT_POST_TEXT;
+                placeholderValues = new String[] { course.getTitle(), course.getTitle(), author.getName(), post.getTitle(), post.getContent(), post.getCreationDate().toString() };
+                notification = new GroupNotification(course, title, text, true, placeholderValues, author, groupNotificationType);
                 notification.setTransientAndStringTarget(createCoursePostTarget(post, course));
             }
+            default -> throw new UnsupportedOperationException("Unsupported NotificationType: " + notificationType);
+        }
+        return notification;
+    }
+
+    /**
+     * Creates an instance of GroupNotification based on the passed parameters.
+     *
+     * @param post                  for which a notification should be created
+     * @param answerPost            to the post for which the notification should be created
+     * @param author                of the notification
+     * @param groupNotificationType user group type the notification should target
+     * @param notificationType      type of the notification that should be created
+     * @param course                the post belongs to
+     * @return an instance of GroupNotification
+     */
+    public static GroupNotification createNotification(Post post, AnswerPost answerPost, User author, GroupNotificationType groupNotificationType,
+            NotificationType notificationType, Course course) {
+        String title;
+        String text;
+        String[] placeholderValues;
+        GroupNotification notification;
+        switch (notificationType) {
             case NEW_REPLY_FOR_EXERCISE_POST -> {
                 Exercise exercise = post.getExercise();
                 title = NEW_REPLY_FOR_EXERCISE_POST_TITLE;
-                text = "Exercise \"" + exercise.getTitle() + "\" got a new reply.";
-                notification = new GroupNotification(course, title, text, author, groupNotificationType);
+                text = NEW_REPLY_FOR_EXERCISE_POST_TEXT;
+                placeholderValues = new String[] { exercise.getTitle(), course.getTitle(), post.getAuthor().getName(), post.getTitle(), post.getContent(),
```

Comment:
The same here, append the different stuff to the array. I am pretty sure you can extract a common part of the array creation into a method or something. You basically duplicate ```kotlin
course.getTitle(), post.getAuthor().getName(), post.getTitle(), post.getContent(),
post.getCreationDate().toString(), answerPost.getAuthor().getName(), answerPost.getContent(), answerPost.getCreationDate().toString()``` three times here.

Reply:
Do we also really need all these values. We have to remember that all values are stored in the database and should not just save everything we can. IMO course-title, postTitle, answerPostContent and answerPostAuthor should be enough for this case

Reply:
If we want to display conversations the way Slack or WhatsApp does I think all the values Tim suggested are necessary.
I extracted setting them in a separate method.